2020/day3.clj

```diff
@@ -7,27 +7,39 @@
        (string/split-lines)))
 
 (defn next-location
-  [loc [right down]]
-  [(+ (first loc) right)
-   (+ (second loc) down)])
+  [[row col] [right down]]
+  [(+ row right)
+   (+ col down)])
 
 (defn locations-s
-  [height [right down]]
-  (take-while #(< (second %) height)
-              (map #(next-location [0 0] [(* right %) (* down %)])
-                   (range))))
+  [slope]
+  (let [right (first slope)
+        down  (second slope)]
+    (map (fn [n]
+           (next-location [0 0] [(* right n) (* down n)]))
+         (range))))
+
+(defn valid-locations
+  [height locations]
+  (->> locations
+       (take-while (fn [[row col]] (< col height)))))
```

Comment:
compressing the fn - why?  also, prefix unused inputs with `_`

Reply:
> compressing the fn - why? also, prefix unused inputs with `_`

```
(defn with-height?
  [[_ col] height]
  (< col height))

(defn valid-locations
  [height locations]
  (->> locations
       (take-while (fn [location] (with-height? location height)))))
```

How was the new code?